CONTRIBUTING.md

```diff
@@ -31,6 +31,10 @@ Git has the `-s` option (lower case) for `commit` that can sign off a commit for
 
 In order to maintain a high software quality standard, we strongly prefer contributions to follow these rules:
 
+- Each commit in a pull request should be atomic, in the sense that it passes all checks (build, static code analysis,
+  tests) individually. Do not add commits to a pull request that (partly) undo changes from an earlier commit in the
```

Comment:
Do you think the second sentence applies also to renaming a previously introduced variable?

Reply:
Depending on what cases you actually have in mind, I might have doubts that this always makes the review harder.

Reply:
> Do you think the second sentence applies also to renaming a previously introduced variable?

No, as I wouldn't consider this as "undoing" in the strict sense. Sometimes it's clearer in the context of recent changes why e.g. a renaming becomes meaningful at that point, and we should allow that IMO.

> (And how about making this a `soft` rule?)

I'd prefer to not go as far as categorizing rules here (which we'd then need to do for *all* rules for consistency), but simply use these rules as the basic guidelines, where each reviewers (i.e. core contributor) has *some* freedom to make exceptions during a review.

Reply:
> Depending on what cases you actually have in mind, I might have doubts that this always makes the review harder, it could e.g. make the review even easier.

I have the case in mind where a review has a significant size (of, say, 10 commits) and thus you decide to review it commit-by-commit, you make same comments to code that was introduced, only to find that code being removed or completely rewritten in a later commit. This is a complete waste of the reviewer's time, which I'd like to avoid.

Reply:
Ok, thanks for these explanations. Makes all sense to me now as well.